paasta_tools/deployd/watchers.py

```diff
@@ -242,7 +243,11 @@ def get_service_instances_with_changed_id(marathon_client, instances, cluster):
                                               instance=instance,
                                               cluster=cluster,
                                               soa_dir=DEFAULT_SOA_DIR)
-        if config.format_marathon_app_dict()['id'] not in marathon_app_ids:
+        try:
+            config_app_id = config.format_marathon_app_dict()['id']
+        except NoDockerImageError:
+            config_app_id = None
+        if not config_app_id or config_app_id not in marathon_app_ids:
```

Comment:
Is this right?

Isn't it "if config_appid" (or config_app_id is not None)


Also this line here about `config_app_id not in marathon_ap_ids`, does that mean we would never give attention to draining apps and stuff? (where the appid is already in place)